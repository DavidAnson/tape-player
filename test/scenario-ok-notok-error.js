// @ts-check

"use strict";

const tape = require("tape");
require("../index.js");

tape("ok test", (test) => {
  test.ok(1);
  test.ok(2, "2 message");
  test.ok(0);
  test.ok(false, "false message");
  test.end();
});

tape("notOk test", (test) => {
  test.notOk(0);
  test.notOk(false, "false message");
  test.notOk(1);
  test.notOk(2, "2 message");
  test.end();
});

tape("error test", (test) => {
  test.error(0);
  test.error(false, "false message");
  test.error(1);
  test.error(2, "2 message");
  const err = Error("error instance");
  err.stack = "stack";
  test.error(err);
  test.error(err, "error message");
  test.end();
});
