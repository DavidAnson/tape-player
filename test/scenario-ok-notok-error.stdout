ok test
  /home/pi/tape-player/test/scenario-ok-notok-error.js:11:8
  Message:  should be truthy
  Operator: ok
  Expected: true
  Actual:   0
  /home/pi/tape-player/test/scenario-ok-notok-error.js:12:8
  Message:  false message
  Operator: ok
  Expected: true
  Actual:   false
notOk test
  /home/pi/tape-player/test/scenario-ok-notok-error.js:19:8
  Message:  should be falsy
  Operator: notOk
  Expected: false
  Actual:   1
  /home/pi/tape-player/test/scenario-ok-notok-error.js:20:8
  Message:  2 message
  Operator: notOk
  Expected: false
  Actual:   2
error test
  /home/pi/tape-player/test/scenario-ok-notok-error.js:27:8
  Message:  1
  Operator: error
  Actual:   1
  /home/pi/tape-player/test/scenario-ok-notok-error.js:28:8
  Message:  2 message
  Operator: error
  Actual:   2
  /home/pi/tape-player/test/scenario-ok-notok-error.js:31:8
  Message:  Error: error instance
  Operator: error
  Actual:   [stack]
  /home/pi/tape-player/test/scenario-ok-notok-error.js:32:8
  Message:  error message
  Operator: error
  Actual:   [stack]

Tests:    3
Asserts:  14
Failures: 8
